### PR TITLE
Enhancement handbook updates - week 0 updates

### PR DESCRIPTION
**Pull request by marosset**
#### What type of PR is this:
/kind cleanup
/kind documentation

#### What this PR does / why we need it:
This PR updates the Enhancements role handbook to reflect the changes away from using Google Sheets for tracking enhancements to GitHub Project Boards.

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/sig-release/issues/2109

#### Special notes for your reviewer:

This PR only addresses content up to and including week 0 responsibilities.
I will update other sections soon.

### CODE REVIEW

File: release-team/role-handbooks/enhancements/README.md
Comment (leonardpahlke) [inline release-team/role-handbooks/enhancements/README.md:142]:
```suggestion
- Add Shadows to the `release-team` and `release-team-enhancements` GitHub teams. (See [access-required](#access-required))
```

**Review by leonardpahlke — APPROVED** [review]
Thanks @marosset! :) 
/hold if someone else likes to take a look

File: release-team/role-handbooks/enhancements/README.md
Comment (Priyankasaggu11929) [reply release-team/role-handbooks/enhancements/README.md:142]: Hey @marosset, just a suggestion (copied from [Leo's addition to the Release Lead Handbook](https://github.com/kubernetes/sig-release/pull/2100/files#diff-176705076aa18b845e0fcb445f9cb129010094bb7ac6e983a5372703d8b8d594R76-R77)).

Maybe we can also add the following bit about the creation of `Enhancements shadow rotation sheet` (for enhancement role's updates in the weekly release team meetings) in Week 0/1 tasks.

```
 [ ] Shadow rotation sheet created for release team meeting enhancements role updates
  - Access: : **restricted access**, edit rights shared with release team enhancements shadows individually
```

**Review by Priyankasaggu11929 — COMMENTED** [review]
Sorry, out of scope of changes for this PR

But should we also rework the following section now, with references to the new GitHub Enhancements Tracking Board?

https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/enhancements#working-with-the-enhancement-tracking-sheet

**Comment by leonardpahlke** [comment]
> But should we also rework the following section now, with references to the new GitHub Enhancements Tracking Board?

> https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/enhancements#working-with-the-enhancement-tracking-sheet

+1

**Comment by marosset** [comment]
> > But should we also rework the following section now, with references to the new GitHub Enhancements Tracking Board?
> 
> > https://github.com/kubernetes/sig-release/tree/master/release-team/role-handbooks/enhancements#working-with-the-enhancement-tracking-sheet
> 
> +1

I was going to make another PR for working with the project board but I can keep updating this PR if everyone wants.
Let me know either way.

**Comment by leonardpahlke** [comment]
we can split it in two PRs 👍

**Comment by salaxander** [comment]
This looks good to me, I think it's safe to unhold

**Comment by marosset** [comment]
/unhold

I'll address extra feedback in the next (few) PRs to the role-handbook :)

**Pull request merged**
